contrib/pwdauth.c

```diff
@@ -223,8 +223,7 @@ main(argc, argv)
 		fprintf(stderr, "Who are you?\n");
 		exit(2);
 	}
-	strncpy(myname, pw->pw_name, sizeof myname - 1);
-	myname[sizeof myname - 1] = '\0';
+	strscpy(myname, pw->pw_name, sizeof myname);
```

Comment:
I see no reason not to check for E2BIG here.

Reply:
I tried to not modify behavior in the same commit.

If we do that, I'd do it in a separate one.

I'm fine with checking truncation, but I don't know in which cases it would be important.  So, if you mark all of them, I can fix them in a continuation commit.

Thanks!

Reply:
Sounds good, thanks, will do.